src/libolm_migration.rs

```diff
@@ -0,0 +1,446 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Migration from libolm to Vodozemac.
+
+use std::{iter, time::Duration};
+
+use anyhow::Context;
+use js_sys::{Array, Date, Uint8Array};
+use matrix_sdk_common::ruma::{
+    DeviceKeyAlgorithm, MilliSecondsSinceUnixEpoch, SecondsSinceUnixEpoch, UInt,
+};
+use matrix_sdk_crypto::{
+    olm::PrivateCrossSigningIdentity,
+    store::{BackupDecryptionKey, Changes, DynCryptoStore, PendingChanges},
+    types::EventEncryptionAlgorithm,
+    Session,
+};
+use matrix_sdk_crypto::vodozemac;
+use matrix_sdk_crypto::vodozemac::{Curve25519PublicKey, Ed25519PublicKey};
+use wasm_bindgen::prelude::*;
+
+use crate::{
+    identifiers::{DeviceId, RoomId, UserId},
+    js::downcast,
+    store::StoreHandle,
+};
+
+type Result<T, E = JsError> = std::result::Result<T, E>;
+
+/// Migration routines
+///
+/// The public methods are exposed as static methods on this class, for namespacing and to enable
+/// easier mocking in unit tests.
+#[wasm_bindgen()]
+#[derive(Debug, Default)]
+pub struct Migration {}
+
+
+/// The base dataset that is important to migrate to the Rust SDK.
+///
+/// Can be imported into the rust store with {@link #migrateBaseData}.
+#[wasm_bindgen(getter_with_clone)]
+#[derive(Debug, Default)]
+pub struct BaseMigrationData {
```

Comment:
Shouldn't there be a list of tracked users here? or a `migrate_tracked_users` method?

Reply:
As discussed elsewhere: I don't think there's any need to migrate the tracked users.